### PR TITLE
components/global: add DialogForm component for displaying forms in a modal

### PR DESCRIPTION
**Pull request by maceksimon**
Adds `DialogForm` component with support for validation before submission (close).
Integrated and tested in `FormFieldCompany` component.
Will be used in `RegisterChallengePage` for creating new company, group etc.

### CODE REVIEW

File: src/components/global/DialogForm.vue [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+      isOpen,
+      close,
+      submit,
+    };
+  },
+});
+</script>
+
+<template>
+  <q-dialog square persistent v-model="isOpen" data-cy="dialog-form">
```
Comment (tmszi) [inline src/components/global/DialogForm.vue:105]: Dialog component should looks like (base style) same across code, and we can reuse one base Dialog component for all use cases. See my comment below.

https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/270efb98d2830ab25af92092282b63e86280fbdc/src/components/global/DialogCard.vue#L77

File: src/components/global/DialogForm.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          class="dialog-close text-grey-10"
+          data-cy="dialog-close"
+        />
+      </q-card-actions>
+    </q-card>
+  </q-dialog>
+</template>
+
+<style lang="scss" scoped>
+.q-dialog__inner > div {
```
Comment (tmszi) [inline src/components/global/DialogForm.vue:158]: Style is duplicated with `DialogCard` component. https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/270efb98d2830ab25af92092282b63e86280fbdc/src/components/global/DialogCard.vue#L132-L148.

**Comment by tmszi** [comment]
Could you resolve conflicts, please?

File: src/components/global/DialogForm.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/global/DialogForm.vue]: Reworked into a `DialogDefault` component.

File: src/components/global/DialogForm.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/global/DialogForm.vue]: Reworked into a `DialogDefault` component.

**Comment by tmszi** [comment]
Should not be refactored `DialogStates` component too or you can do it in the next PR? https://github.com/auto-mat/ride-to-work-by-bike-frontend/blob/25689bdab2657acb1e79722a080ed152372c719d/src/components/global/DialogStates.vue#L88-L167

**Comment by maceksimon** [comment]
I would rather do it in separate PR. This one is getting large.

**Review by tmszi — APPROVED** [review]

**Pull request merged**